records/generator_test.go

```diff
@@ -317,6 +324,48 @@ func TestInsertState(t *testing.T) {
 
 }
 
+// ensure that container IP are returned, taken from the docker inspect output in the state
+func TestInsertStateWithDockerInspectParsing(t *testing.T) {
+	var sj StateJSON
+
+	b, err := ioutil.ReadFile("../factories/fake.json")
+	if err != nil {
+		t.Fatal("missing test data")
+	}
+
+	err = json.Unmarshal(b, &sj)
+	if err != nil {
+		t.Fatal(err)
+	}
+	sj.Leader = "master@144.76.157.37:5050"
+
+	masters := []string{"144.76.157.37:5050"}
+	spec := labels.ForRFC952()
+	rg := &RecordGenerator{}
+	rg.InsertState(sj, "mesos", "mesos-dns.mesos.", "127.0.0.1", masters, spec, true)
+
+	rrs, ok := rg.As["liquor-store.marathon.mesos."]
+	if !ok {
+		t.Error("should find this running task - A record")
+	}
+	if got, want := rrs, []string{"1.2.3.11", "1.2.3.12"}; !reflect.DeepEqual(got, want) {
+		t.Errorf("should return the slave ips 1.2.3.11, 1.2.3.12 for the task, but got %v", rrs)
+	}
+
+	rrs, ok = rg.As["_container.liquor-store.marathon.mesos."]
+	if !ok {
+		t.Error("should find the container ip because ParseDockerInspectForContainerIP is enabled")
+	}
+	if len(rrs) != 1 || rrs[0] != "10.3.0.1" {
```

Comment:
If we're now using `reflect.DeepEqual` in the previous case, we should use it in the others too.